elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLProjectionInfo.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2019, Oath Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.graphql.parser;
+
+import com.yahoo.elide.request.EntityProjection;
+import com.yahoo.elide.request.Relationship;
+
+import graphql.language.SourceLocation;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A helper class that contains a collection of root-level entity projections and relationship map constructed from
+ * a {@link graphql.language.Document}.
+ */
+@AllArgsConstructor
+public class GraphQLProjectionInfo {
+    @Getter
+    private final SourceLocation rootLocation;
+
+    @Getter
+    private final Collection<EntityProjection> projections;
+
+    @Getter
+    private final Map<SourceLocation, Relationship> relationshipMap;
```

Comment:
I'm not sure you need all this if you use recursion to resolve the EntityProjection given the current sourceLocation.

Reply:
This is for getting EntityProjection from the `GraphQLProjectionInfo` only. When the request goes through the `GraphQL` core code, a new `document` would be generated and parsed by `GraphQL` module itself, which would contain the same content as the one we parsed but it would be a different object. And the graphql `context` would contain the graphql object generated by `GraphQL` module. So I need this `SourceLocation` to get the projection I generated.